lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/addressing/AdHocMessageRoutingBuilder.java

```diff
@@ -64,106 +99,126 @@ private MessageRouting build(DestinationAddressContainer dac) {
      * @param adHocChannel specific ad hoc channel {@link AdHocChannel}
      * @return this builder
      */
-    public AdHocMessageRoutingBuilder viaChannel(AdHocChannel adHocChannel) {
+    public AdHocMessageRoutingBuilder channel(AdHocChannel adHocChannel) {
+        Validate.isTrue(!channelChanged);
```

Comment:
Don't forget about the message.

`Validate.isTrue(!channelChanged, "Channel has already been set.");`